### PR TITLE
add documentation for system.metric_log table

### PR DESCRIPTION
**Pull request by Slach**
Signed-off-by: Slach <bloodjazman@gmail.com>

I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Documentation (changelog entry is not required)

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
add documentation to `system.metric_log` table

### CODE REVIEW

File: docs/en/operations/system_tables.md [hunk 1]
```diff
@@ -273,10 +275,68 @@ SELECT * FROM system.metrics LIMIT 10
...
 **See Also**
 
 - [system.asynchronous_metrics](#system_tables-asynchronous_metrics) — Contains periodically calculated metrics.
 - [system.events](#system_tables-events) — Contains a number of events that occurred.
+- [system.metric_log](#system_tables-metric_log) — Contains a history of metrics values from tables `system.metrics` и `system.events`.
+- [Monitoring](monitoring.md) — Base concepts of ClickHouse monitoring.
+
+## system.metric_log {#system_tables-metric_log}
+
+Contains history of metrics values from tables `system.metrics` и `system.events`, periodically flush to disc with MegreTree() table engine
```
Comment (alexey-milovidov) [inline docs/en/operations/system_tables.md]: Usually you write dis*c* if it's round like compact disc and dis*k* if you mean some storage media like HDD and SSD.

**Review by alexey-milovidov — APPROVED** [review]

File: docs/en/operations/system_tables.md (hunk 1 above)
Comment (Slach) [reply docs/en/operations/system_tables.md]: thanks, fixed

Comment (alexey-milovidov) [reply docs/en/operations/system_tables.md]: The table is not necessarily MergeTree.

File: docs/en/operations/system_tables.md [hunk 2]
```diff
@@ -273,10 +275,68 @@ SELECT * FROM system.metrics LIMIT 10
...
 
 - [system.asynchronous_metrics](#system_tables-asynchronous_metrics) — Contains periodically calculated metrics.
 - [system.events](#system_tables-events) — Contains a number of events that occurred.
+- [system.metric_log](#system_tables-metric_log) — Contains a history of metrics values from tables `system.metrics` и `system.events`.
+- [Monitoring](monitoring.md) — Base concepts of ClickHouse monitoring.
+
+## system.metric_log {#system_tables-metric_log}
+
+Contains history of metrics values from tables `system.metrics` and `system.events`, periodically flushed to disk.
+For turn on metrics history collection on `system.metric_log`, create `/etc/clickhouse-server/config.d/metric_log.xml` with following content:
```
Comment (qoega) [inline docs/en/operations/system_tables.md]: Probably
```suggestion
To turn on metrics history collection on `system.metric_log`, create `/etc/clickhouse-server/config.d/metric_log.xml` with following content:
```
or even
```suggestion
Metrics history collection is enabled in server settings.
For example create `/etc/clickhouse-server/config.d/metric_log.xml` with following content or add this block to other server configuration file:
```

File: docs/en/operations/system_tables.md [hunk 3]
```diff
@@ -273,10 +275,68 @@ SELECT * FROM system.metrics LIMIT 10
...
+```xml
+<yandex>
+    <metric_log>
+        <database>system</database>
+        <table>metric_log</table>
+        <flush_interval_milliseconds>7500</flush_interval_milliseconds>
+        <collect_interval_milliseconds>1000</collect_interval_milliseconds>
```
Comment (qoega) [inline docs/en/operations/system_tables.md]: Probably there should be TTL or some depth in this table configuration or size would grow without a limit

**Review by qoega — CHANGES_REQUESTED** [review]

File: docs/en/operations/system_tables.md (hunk 2 above)
Comment (Slach) [reply docs/en/operations/system_tables.md]: system.metric_log doesn't enable by default, so a first suggestion is chosen

File: docs/en/operations/system_tables.md (hunk 3 above)
Comment (Slach) [reply docs/en/operations/system_tables.md]: @qoega 
I agree, could you suggest code changes in 
https://github.com/ClickHouse/clickhouse/tree/master/dbms/src/Interpreters/MetricLog.h
https://github.com/ClickHouse/clickhouse/tree/master/dbms/src/Interpreters/MetricLog.cpp
for creating MergeTree table with TTL?

Comment (Slach) [reply docs/en/operations/system_tables.md]: as i see in https://github.com/ClickHouse/ClickHouse/blob/master/dbms/src/Interpreters/SystemLog.cpp#L27

currently, we don't have the ability to use TTL for system tables

**Comment by alexey-milovidov** [comment]
@Slach You can specify TTL for system tables, see https://github.com/ClickHouse/ClickHouse/pull/8794
@qoega But metric_log is very small table, it's unlikely that anyone will ever need TTL for it.

**Pull request merged**